src/vivarium_nih_us_cvd/components/healthcare_visits.py

```diff
@@ -118,12 +162,10 @@ def on_initialize_simulants(self, pop_data: SimulantData) -> None:
             df[models.MYOCARDIAL_INFARCTION_MODEL_NAME]
             == models.ACUTE_MYOCARDIAL_INFARCTION_STATE_NAME
         )
-        emergency_not_already_scheduled = index[
-            (mask_acute_is | mask_acute_mi) & ~(mask_chronic_is | mask_post_mi)
-        ]
-        visitors[data_values.VISIT_TYPES.EMERGENCY] = emergency_not_already_scheduled
-        df.loc[emergency_not_already_scheduled] = self.visit_doctor(
-            df=df.loc[emergency_not_already_scheduled],
+        emergency = index[(mask_acute_is | mask_acute_mi)]
+        visitors[data_values.VISIT_TYPE.EMERGENCY] = emergency
```

Comment:
Why don't you just set the visit type column here rather than doing this complicate passing around of dictionaries? Then you can just pass the index of the simulants who are attending the doctor to this function.

Reply:
Will confirm with Syl that she wants visitors with missed appointments to also be part of background random draws.